frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageParseDialog.vue

```diff
@@ -0,0 +1,449 @@
+                    @insert-below="insertNewIngredient(index + 1)"
+                  />
+                  <p class="pt-0 pb-4 my-0 text-caption">
+                    {{ $t("recipe.original-text-with-value", { originalText: ingredient.input }) }}
+                  </p>
+                </div>
+              </TransitionGroup>
+            </VueDraggable>
+          </v-expansion-panel-text>
+        </v-expansion-panel>
+      </v-expansion-panels>
+    </v-container>
+    <template v-if="!state.loading.parser" #custom-card-action>
+      <BaseButton
+        v-if="!state.allReviewed"
+        color="info"
+        :icon="$globals.icons.arrowRightBold"
+        icon-right
+        :text="$t('general.next')"
+        @click="nextIngredient"
+      />
+      <BaseButton
+        v-else
+        create
+        :text="$t('general.save')"
+        :icon="$globals.icons.save"
+        :loading="state.loading.save"
+        @click="saveIngs"
+      />
+    </template>
+  </BaseDialog>
+</template>
+
+<script setup lang="ts">
+import { VueDraggable } from "vue-draggable-plus";
+import type { IngredientFood, IngredientUnit, ParsedIngredient, RecipeIngredient } from "~/lib/api/types/recipe";
+import type { Parser } from "~/lib/api/user/recipes/recipe";
+import type { NoUndefinedField } from "~/lib/api/types/non-generated";
+import { useAppInfo, useUserApi } from "~/composables/api";
+import { parseIngredientText } from "~/composables/recipes";
+import { useFoodData, useFoodStore, useUnitData, useUnitStore } from "~/composables/store";
+import { useGlobalI18n } from "~/composables/use-global-i18n";
+import { alert } from "~/composables/use-toast";
+import { useParsingPreferences } from "~/composables/use-users/preferences";
+
+const props = defineProps<{
+  modelValue: boolean;
+  ingredients: NoUndefinedField<RecipeIngredient[]>;
+}>();
+
+const emit = defineEmits<{
+  (e: "update:modelValue", value: boolean): void;
+  (e: "save", value: NoUndefinedField<RecipeIngredient[]>): void;
+}>();
+
+const i18n = useGlobalI18n();
+const api = useUserApi();
+const appInfo = useAppInfo();
+const drag = ref(false);
+
+const unitStore = useUnitStore();
+const unitData = useUnitData();
+const foodStore = useFoodStore();
+const foodData = useFoodData();
+
+const parserPreferences = useParsingPreferences();
+const parser = ref<Parser>(parserPreferences.value.parser || "nlp");
+const availableParsers = computed(() => {
+  return [
+    {
+      text: i18n.t("recipe.parser.natural-language-processor"),
+      value: "nlp",
+    },
+    {
+      text: i18n.t("recipe.parser.brute-parser"),
+      value: "brute",
+    },
+    {
+      text: i18n.t("recipe.parser.openai-parser"),
+      value: "openai",
+      hide: !appInfo.value?.enableOpenai,
+    },
+  ];
+});
+
+/**
+ * If confidence of parsing is below this threshold,
+ * we will prompt the user to review the parsed ingredient.
+ */
+const confidenceThreshold = 0.85;
```

Comment:
The confidence threshold is hardcoded here. There's an opportunity to make this configurable if we want.